### ISSUE TITLE
Question: Do services store exactly one token?

### ISSUE THREAD

**Issue by poisa** [beaulebens/keyring#33]
Quick question on the token store design:

While getting the tokens from the store you get an array of `Keyring_Access_Token`. Can we assume that if you have successfully connected to a service and ask the token store for that specific service we will be getting just _one_ token? 

That is to say, if we do:

```php
$tokens = $tokenStore->get_tokens( [ 'service' => 'google-analytics' ] );
```

... then `$tokens` will **always** be an array of exactly 1 element? (or 0 if we've never connected to that service).

**Comment by beaulebens** [comment]
If you've only connected to that service once, then you should only ever get one token back, but if you ask for tokens based on a service, you should be ready to handle receiving multiple tokens back. If you just want a specific one, then you should query more specifically -- either by a unique ID for a specific token, by including a user ID in the query, or something else.

The structure is intended to allow for multiple connections, per user, per service though out of the box.

**Comment by poisa** [comment]
Cool, thanks for the prompt response. 

> either by a unique ID for a specific token, by including a user ID in the query, or something else.

Yeah, I saw that if you don't pass a user it defaults to the logged-in user which in my case is as specific as I need to be.

By the way I just hovered over your avatar and saw that you are an Automattician! It makes sense now that I think of it given that my recently started Automattic trial involves this very same repo! Hence my _(cough, cough)_ pull request. 

(shameless... I know :stuck_out_tongue_closed_eyes:)
